tables/nlb_access_log/nlb_access_log.go

```diff
@@ -0,0 +1,162 @@
+package nlb_access_log
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/turbot/tailpipe-plugin-sdk/schema"
+)
+
+type NlbAccessLog struct {
+	schema.CommonFields
+
+	ALPNBEProtocol            string    `json:"alpn_be_protocol,omitempty"`
+	ALPNClientPreferenceList  []string  `json:"alpn_client_preference_list,omitempty"`
+	ALPNFEProtocol            string    `json:"alpn_fe_protocol,omitempty"`
+	ChosenCertArn             string    `json:"chosen_cert_arn,omitempty"`
+	ChosenCertSerial          string    `json:"chosen_cert_serial,omitempty"`
+	ClientIP                  string    `json:"client_ip,omitempty"`
+	ClientPort                int       `json:"client_port,omitempty"`
+	ConnectionTime            int       `json:"connection_time,omitempty"`
+	DestinationIP             string    `json:"destination_ip,omitempty"`
+	DestinationPort           int       `json:"destination_port,omitempty"`
+	DomainName                string    `json:"domain_name,omitempty"`
+	Elb                       string    `json:"elb"`
+	IncomingTLSAlert          string    `json:"incoming_tls_alert,omitempty"`
+	Listener                  string    `json:"listener,omitempty"`
+	ReceivedBytes             int64     `json:"received_bytes,omitempty"`
+	SentBytes                 int64     `json:"sent_bytes,omitempty"`
+	TLSCipher                 string    `json:"tls_cipher,omitempty"`
+	TLSConnectionCreationTime time.Time `json:"tls_connection_creation_time"`
+	TLSHandshakeTime          int       `json:"tls_handshake_time,omitempty"`
+	TLSNamedGroup             string    `json:"tls_named_group,omitempty"`
+	TLSProtocolVersion        string    `json:"tls_protocol_version,omitempty"`
+	Timestamp                 time.Time `json:"timestamp"`
+	Type                      string    `json:"type"`
+	Version                   string    `json:"version"`
+}
+
+// InitialiseFromMap - initialise the struct from a map
+func (l *NlbAccessLog) InitialiseFromMap(m map[string]string) error {
+	var err error
+	ISO8601 := "2006-01-02T15:04:05"
+	for key, value := range m {
+		if value == "-" {
+			continue
+		}
+		switch key {
+		case "timestamp":
+			ts, err := time.Parse(ISO8601, value)
+			if err != nil {
+				return fmt.Errorf("error parsing timestamp: %w", err)
+			}
+			l.Timestamp = ts
+		case "type":
+			l.Type = value
+		case "version":
+			l.Version = value
+		case "elb":
+			l.Elb = value
+		case "listener":
+			l.Listener = value
+		case "incoming_tls_alert":
+			l.IncomingTLSAlert = value
+		case "chosen_cert_serial":
+			l.ChosenCertSerial = value
+		case "client":
+			parts := strings.Split(value, ":")
+			l.ClientIP = parts[0]
+			l.ClientPort, err = strconv.Atoi(parts[1])
```

Comment:
Instead of doing this logic here, is it not possible to use `$clientip:$clientport` in your pattern to obtain these values individually when parsing?

Reply:
Only valid if we ALWAYS get values!